src/spatial/node/Accumulator.scala

```diff
@@ -1,5 +1,42 @@
 package spatial.node
 
-class Accumulator {
+import argon._
+import forge.tags._
+import spatial.lang._
 
+sealed abstract class Accum
+object Accum {
+  case object Add extends Accum
+  case object Mul extends Accum
+  case object Min extends Accum
+  case object Max extends Accum
 }
+
+abstract class RegAccum[A:Bits] extends Accumulator[A] {
+  def bank = Nil
+  def ofs = Nil
+}
+
+
+@op case class RegAccumFMA[A:Bits](
+    mem:  Reg[A],
+    m0:   Bits[A],
+    m1:   Bits[A],
+    en:   Set[Bit],
+    first: Bit)
+  extends RegAccum[A]
+
+@op case class RegAccumOp[A:Bits](
+    mem: Reg[A],
+    in:  Bits[A],
+    en:  Set[Bit],
+    op:  Accum,
+    first: Bit)
+  extends RegAccum[A]
+
```

Comment:
Minor housekeeping detail, but does it make sense to just have RegAccumLambda1/2 and RegAccumOp1/2 and have FMA be another case object?

Reply:
Yeah we could structure it that way. I had no other use cases requiring 2 inputs at the moment, so I just have it as FMA for now. If there are more (or we find more) we can generalize.